ZuluContent/Spells/Base/Spell.cs

```diff
@@ -409,31 +329,21 @@ public virtual TimeSpan GetCastRecovery()
 
         public virtual TimeSpan GetCastDelay()
         {
-            if (IsWand())
-                return TimeSpan.Zero;
-
-            var baseDelay = CastDelayBase.TotalMilliseconds;
+            var delay = (double)Circle.Delay;
 
-            var fcDelay = -(CastDelayFastScalar * CastDelaySecondsPerTick);
-
-            //int delay = CastDelayBase + circleDelay + fcDelay;
-            var delay = baseDelay + fcDelay;
-
-            Caster.FireHook(h => h.OnGetCastDelay(Caster, this, ref delay));
-
-            return delay < CastDelayMinimum.TotalMilliseconds
-                ? CastDelayMinimum
-                : TimeSpan.FromSeconds(delay);
+            if (Circle > 8)
```

Comment:
FYI this looks weird but is correct, I've done side-by-side comparisons with POL93 ZH3 to make sure. 

Basically the core does the casting checks for the normal spellbook spells which doesn't have any modifications to the cast delays. When ebook/nbook spells are cast they go via the `TryToCast` function in `spelldata.inc` which is where the delay is shortened for Mages. Technically it should also be longer for Warriors but it seemed kinda redundant so I haven't included it.